gqlalchemy/query_builder.py

```diff
@@ -139,15 +146,46 @@ def construct_query(self) -> str:
 
 
 class WhereConditionPartialQuery(PartialQuery):
-    def __init__(self, keyword: str, query: str):
-        super().__init__(DeclarativeBaseTypes.WHERE)
+    def __init__(self, item: str, operator: str, keyword: str = None, **kwargs):
+        super().__init__(type=DeclarativeBaseTypes.WHERE)
 
         self.keyword = keyword
-        self.query = query
+        self._build_where_query(item=item, operator=operator, **kwargs)
```

Comment:
If this method is called in init and not in construct query, the type checks are done before execution?

Reply:
Yes, the plan is to validate the query at a construction time